bolt-sidecar/src/telemetry/metrics.rs

```diff
@@ -81,6 +86,25 @@ impl ApiMetrics {
         counter!(INCLUSION_COMMITMENTS_ACCEPTED).increment(1);
     }
 
+    pub fn increment_gross_tip_revenue(mut tip: u128) {
+        // If the tip is too large, we need to split it into multiple u64 parts
+        if tip > u64::MAX as u128 {
+            let mut parts = Vec::new();
+            while tip > u64::MAX as u128 {
+                parts.push(u64::MAX);
+                tip -= u64::MAX as u128;
+            }
+
+            parts.push(tip as u64);
+
+            for part in parts {
+                counter!(GROSS_TIP_REVENUE).increment(part);
+            }
+        } else {
+            counter!(GROSS_TIP_REVENUE).increment(tip as u64);
+        }
```

Comment:
In my opinion this is a bit simpler/succint and doesn't distinguish between edge cases and regular cases which is nice.
```suggestion
         // Takes also into account tips larger than a u128 by splitting them into u64 parts
        let mut parts = Vec::new();
        let u64_max = u64::MAX as u128;

        while tip > 0 {
            // We can only add a max tip of u64::MAX.
            let min = tip.min(u64_max) as u64;
            parts.push(min);
            // Subtract the part we just added, if something is left then tip is going to be
            // greater than zero.
            tip = tip.saturating_sub(u64_max)
        }

        parts.push(tip as u64);

        for part in parts {
            counter!(GROSS_TIP_REVENUE).increment(part);
        }
```

Reply:
True, but this always does an allocation

Reply:
That's fair, also because with an u64 you can express a tip up to 18.44 ETH :D